Java/core/src/main/java/com/amazon/randomcutforest/state/store/LeafStoreMapper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.state.store;
+
+import java.util.Arrays;
+
+import com.amazon.randomcutforest.IStateMapper;
+import com.amazon.randomcutforest.store.LeafStore;
+
+public class LeafStoreMapper implements IStateMapper<LeafStore, LeafStoreState> {
+    @Override
+    public LeafStore toModel(LeafStoreState state, long seed) {
+        int capacity = state.getPointIndex().length;
+        int[] pointIndex = Arrays.copyOf(state.getPointIndex(), capacity);
+        short[] parentIndex = Arrays.copyOf(state.getParentIndex(), capacity);
+        int[] mass = Arrays.copyOf(state.getMass(), capacity);
+
+        short freeIndexPointer = (short) (state.getFreeIndexes().length - 1);
+        short[] freeIndexes = new short[capacity];
+        System.arraycopy(state.getFreeIndexes(), 0, freeIndexes, 0, freeIndexPointer + 1);
```

Comment:
minor. using `state.getFreeIndexes().length` for the last parameter might be clearer.

Reply:
Done.